_posts/2023-02-22-dependency-api-deprecation.md

```diff
@@ -0,0 +1,55 @@
+---
+title: RubyGems.org Dependency API Deprecation
+layout: post
+author: Samuel Giddins
+author_email: segiddins@segiddins.me
+---
+
+We're deprecating [the Dependency API](https://guides.RubyGems.org/rubygems-org-api/#misc-methods) on RubyGems.org. To learn more, read on, or see the original [RFC][rfc] about this deprecation.
+
+[rfc]: https://github.com/segiddins/rfcs/blob/segiddins/deprecate-dependency-api/0000-deprecate-dependency-api.md
+
+### What is the Dependency API?
+
+The dependency API was the primary way Bundler fetched dependency info for Gemfile resolution from [Bundler 1.1](https://github.com/rubygems/bundler/blob/master/CHANGELOG.md#11pre-january-21-2011) until the release of [Bundler 1.12](https://bundler.io/blog/2016/04/28/the-new-index-format-fastly-and-bundler-1-12.html) in April 2016, which introduced the ["new" Compact Index API](https://andre.arko.net/2014/03/28/the-new-rubygems-index-format/).
+
+The API has a single endpoint, `/api/v1/dependencies` (returning responses as Marshal or as JSON), accepting a single comma-separated `gems` query parameter.
+
+For example, <https://index.RubyGems.org/api/v1/dependencies.json?gems=rails> returns every dependency for every version of the gem `rails`.
+
+### Why is the Dependency API being deprecated?
+
+The dependency API is extremely resource-intensive: even today, it makes up 20-25% of the traffic served by the RubyGems.org Rails app that cannot be cached by Fastly. Each unique list of gems requires RubyGems.org to run a complex database query, and a majority of requests include a unique list of gems.
+
+In contrast to the Dependency API, the Compact Index API is designed to be cachable. Since the  Compact Index has been out for almost seven years, and since Bundler & RubyGems both support fallback to the slower "full" index when the Dependency API is unavailable, we plan to deprecate and delete the dependency API from RubyGems.org over the coming months.
+
+### What impact will this deprecation have?
+
+We expect this deprecation to have no user-facing impact.
+
+If you are installing gems from RubyGems.org using versions of Bundler more than 7 years old (1.11 or older), your installs will get slower. Hopefully, everyone has upgraded their copy of Bundler at least once in the last 7 years, and their installs can continue at full speed.
+
+If you are installing gems from other gem sources, like [Gemstash][1], [Geminabox][2], [Nexus][3], [Artifactory][4], or [GitHub Packages][5], nothing will change. Bundler will continue to try the Compact Index API first, then fall back to the Dependency API, then fall back to the full index.
```

Comment:
unfortunately, change https://issues.sonatype.org/browse/NEXUS-38452?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedId=1257013#comment-1257013

Nexus and artifactory didn't implement compact API, with some tweaks they will be worked in full index mode.